controlplane/kubeadm/internal/cluster.go

```diff
@@ -122,10 +122,10 @@ type healthCheck func(context.Context) (healthCheckResult, error)
 // It does some additional validation to make sure there is a 1;1 match between nodes and machines.
 func (m *ManagementCluster) healthCheck(ctx context.Context, check healthCheck, clusterKey types.NamespacedName, controlPlaneName string) error {
 	nodeChecks, err := check(ctx)
+	errorList := []error{}
```

Comment:
move this up one line?

Reply:
It can also be defined as `var errorList []error` since append will handle nil appropriately.